src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -1010,77 +870,379 @@ private Assembly LoadAssemblyHelper(string assemblyName)
+                    }
+                }
+            }
+            else if (String.Equals(ParameterSetName, "FromMember", StringComparison.OrdinalIgnoreCase))
+            {
+                _sourceCode = GenerateTypeSource(Namespace, Name, _sourceCode, Language);
 
-                    // Ignore some specified reference assemblies
-                    string fileName = PathType.GetFileName(resolvedAssemblyPath);
-                    if (s_refAssemblyNamesToIgnore.Value.Contains(fileName))
+                var sourceText = SourceText.From(_sourceCode);
+                _syntaxTrees.Add(CSharpSyntaxTree.ParseText(sourceText, _csharpParseOptions));
+            }
+            else if (String.Equals(ParameterSetName, "FromSource", StringComparison.OrdinalIgnoreCase))
+            {
+                var sourceText = SourceText.From(_sourceCode);
+                _syntaxTrees.Add(CSharpSyntaxTree.ParseText(sourceText, _csharpParseOptions));
+            }
+            else
+            {
+                Diagnostics.Assert(false, "Invalid parameter set: {0}", this.ParameterSetName);
+            }
+
+            if (CompileOnly)
+            {
+                CSharpCompileToAssembly();
+            }
+            else
+            {
+                // if the source code was already compiled and loaded and not changed
+                // we get the assembly from the cache.
+                if (isSourceCodeUpdated(out Assembly assembly))
+                {
+                    CSharpCompileToAssembly();
+                }
+                else
+                {
+                    if (PassThru)
                     {
-                        WriteVerbose(StringUtil.Format(AddTypeStrings.ReferenceAssemblyIgnored, resolvedAssemblyPath));
-                        continue;
+                        WriteTypes(assembly);
                     }
-                    tempReferences.Add(MetadataReference.CreateFromFile(resolvedAssemblyPath));
+
+                    WriteVerbose(AddTypeStrings.AlreadyCompiledandLoaded);
                 }
-                references = tempReferences.ToArray();
             }
+        }
+
+        private void VisualBasicSourceCodeProcessing()
+        {
+            if (ExtendedOptions != null)
+            {
+                var arguments = VisualBasicDefaultParse(args: ExtendedOptions);
+
+                HandleCompilerErrors(arguments.Errors);
+
+                _visualbasicParseOptions = arguments.ParseOptions;
+                _visualbasicCompilationOptions = arguments.CompilationOptions.WithOutputKind(OutputAssemblyTypeToOutputKind(OutputType));
+
+                if (WarnAsError)
+                {
+                    _visualbasicCompilationOptions = arguments.CompilationOptions.WithGeneralDiagnosticOption(s_generalDiagnosticOption);
+                }
+
+                _emitOptions = arguments.EmitOptions;
```

Comment:
Similar comments as in `CSharpSourceCodeProcessing`.

Reply:
Fixed.